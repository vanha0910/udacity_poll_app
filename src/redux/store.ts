import { combineReducers, configureStore } from '@reduxjs/toolkit'
import authenticationReducer from './auth/authenticationSlice'
import questionReducer from './question/questionsSlice'

const rootReducer = combineReducers({
  authentication: authenticationReducer,
  questions: questionReducer
})
export function setupStore(preloadedState?: Partial<RootState>) {
  return configureStore({
    reducer: rootReducer,
    preloadedState
  })
}
export type RootState = ReturnType<typeof rootReducer>
export type AppStore = ReturnType<typeof setupStore>
export type AppDispatch = AppStore['dispatch']