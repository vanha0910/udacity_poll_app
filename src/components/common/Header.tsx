import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { logoutUser } from "../../redux/auth/authenticationSlice";
import { Link, useNavigate , useLocation} from "react-router-dom";
import { PATH } from "../../constansts";
import { RootState } from "@/redux/store";

const Header = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const location = useLocation();
  const { authedUser }= useSelector((state: RootState) => state.authentication);

  const handleLogout = () => {
    dispatch(logoutUser());
    navigate(PATH.LOGIN)
  };


  return (
    <header className="header">
      <nav className="nav">
        <ul className="nav-list">
          <li  className={`nav-item ${location.pathname === PATH.HOME ? ' active': ''}`}>
            <Link to="/" className={"nav-link"} data-testid="nav-1">
              Home
            </Link>
          </li>
          <li  className={`nav-item ${location.pathname === PATH.LEADER_BOARD ? ' active': ''}`}>
            <Link to="/leaderboard"  className={"nav-link"}  data-testid="nav-2">
              Leaderboard
            </Link>
          </li>
          <li  className={`nav-item ${location.pathname === PATH.NEW_QUESTION ? ' active': ''}`}>
            <Link to="/add"   className={"nav-link"}  data-testid="nav-3">
              New
            </Link>
          </li>
        </ul>
      </nav>
      <div className="user-info">
        {authedUser && (
          <>
            <img
              src={authedUser.avatarURL}
              alt={`${authedUser.name} Avatar`}
              className="user-avatar"
            />
            <span className="user-name">{authedUser.name}</span>
          </>
        )}
        <button className="logout-btn" onClick={handleLogout} data-testid="logout-btn">
          Logout
        </button>
      </div>
    </header>
  );
};

export default Header;
