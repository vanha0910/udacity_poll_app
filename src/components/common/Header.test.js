import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import store from "../../redux";
import Header from "./Header";

test("renders Header component correctly", () => {
  const { container } = render(
    <MemoryRouter>
      <Provider store={store}>
        <Header />
      </Provider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
