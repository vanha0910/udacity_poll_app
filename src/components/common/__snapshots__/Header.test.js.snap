// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <nav
      class="nav"
    >
      <ul
        class="nav-list"
      >
        <li
          class="nav-item  active"
        >
          <a
            class="nav-link"
            data-testid="nav-1"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item "
        >
          <a
            class="nav-link"
            data-testid="nav-2"
            href="/leaderboard"
          >
            Leaderboard
          </a>
        </li>
        <li
          class="nav-item "
        >
          <a
            class="nav-link"
            data-testid="nav-3"
            href="/add"
          >
            New
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="user-info"
    >
      <button
        class="logout-btn"
        data-testid="logout-btn"
      >
        Logout
      </button>
    </div>
  </header>
</div>
`;

exports[`renders Header component correctly 1`] = `
<div>
  <header
    class="header"
  >
    <nav
      class="nav"
    >
      <ul
        class="nav-list"
      >
        <li
          class="nav-item  active"
        >
          <a
            class="nav-link"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item "
        >
          <a
            class="nav-link"
            href="/leaderboard"
          >
            Leaderboard
          </a>
        </li>
        <li
          class="nav-item "
        >
          <a
            class="nav-link"
            href="/add"
          >
            New
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="user-info"
    >
      <button
        class="logout-btn"
      >
        Logout
      </button>
    </div>
  </header>
</div>
`;

exports[`renders MyComponent correctly 1`] = `
<div>
  <header
    class="header"
  >
    <nav
      class="nav"
    >
      <ul
        class="nav-list"
      >
        <li
          class="nav-item  active"
        >
          <a
            class="nav-link"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item "
        >
          <a
            class="nav-link"
            href="/leaderboard"
          >
            Leaderboard
          </a>
        </li>
        <li
          class="nav-item "
        >
          <a
            class="nav-link"
            href="/add"
          >
            New
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="user-info"
    >
      <button
        class="logout-btn"
      >
        Logout
      </button>
    </div>
  </header>
</div>
`;
